stock_picking_auto_create_lot/models/stock_move_line.py

```diff
@@ -37,3 +39,25 @@ def set_lot_auto(self):
             line.lot_id = lot
             if lot.product_id.tracking == "serial":
                 lots_by_product[line.product_id.id] -= lot
+
+    def _action_done(self):
```

Comment:
This approach is a bit cumbersome and creates gaps in serial numbers.  Can we instead change `_set_auto_lot()` of stock.picking as below?

```.py
    def _set_auto_lot(self):
        """
        Allows to be called either by button or through code
        """
        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
        domain = [
            ("picking_id", "in", pickings.ids),
            ("lot_id", "=", False),
            ("lot_name", "=", False),
            ("product_id.tracking", "!=", "none"),
            ("product_id.auto_create_lot", "=", True),
        ]
        # i.e. There is a move line with done quantity in the picking.
        # This is to prevent unnecessary generation of unused serials in case of partial receipts of serial-tracked products.
        if not pickings._check_immediate():
            domain = expression.AND([domain, [("qty_done", ">", 0)]])
        lines = self.env["stock.move.line"].search(domain)
        lines.set_lot_auto()
```

Reply:
Thanks for the suggestion. It is more valid. I updated the code.